include/zephyr/llext/loader.h

```diff
@@ -95,6 +133,11 @@ struct llext_loader {
 	 */
 	void (*finalize)(struct llext_loader *ldr);
 
+	/**
+	 * @brief Storage type of the underlying data accessed by this loader.
+	 */
+	enum llext_storage_type storage;
```

Comment:
It seems to me this should rather go to `struct llext_load_param`

Reply:
As discussed in Discord, IMO the way the data is handled must be strictly tied to the loader.
 
Consider the case of `fs_loader`, for example. It is implicitly TEMPORARY - there is no data that can be read long-term from its storage - even though it might even support `peek` in the future (say, the file is available in some cache, perhaps!). 

If you move this field into `llext_load_param`, people will try to use `fs_loader` with `WRITABLE`. It makes no sense and it should fail, but how do you test for this specific case _inside `llext_load`_? And how do you support other app-specific loader implementations?

Reply:
@pillo79 wouldn't it be possible for the user to select a wrong macro with your approach too? Also we can define several macros for the parameters to extend `LLEXT_LOAD_PARAM_DEFAULT`

Reply:
> @pillo79 wouldn't it be possible for the user to select a wrong macro with your approach too?

Well, the default initializer for `fs_loader` sets the storage to TEMPORARY, no way around it unless you actually fill the loader struct by yourself. And the different default initializers for BUF_LOADER make any mistake pretty easy to spot.

> Also we can define several macros for the parameters to extend `LLEXT_LOAD_PARAM_DEFAULT`

The whole point of this PR is to unambiguously state the storage type of the data exposed by each loader. Moving this to an optional struct defeats the purpose IMO (and my question above would still apply).

Reply:
> The whole point of this PR is to unambiguously state the storage type of the data exposed by each loader. Moving this to an optional struct defeats the purpose IMO (and my question above would still apply).

@pillo79 ok, I rest my case :-) The point - as you say - seems to be, that loader parameters are optional and we really want an explicit selection of a loader type to be compulsory. Looks a bit confusing IMHO to have "parameters" in two separate locations, but if others are fine with this - I can live with this too.